### ISSUE TITLE
RazerS3 misses matches and gives inconsistent results

### ISSUE THREAD

**Issue by sarahet** [seqan/seqan#2111]
I am working a lot with RazerS3 lately and I came along the following problem: I am working with short reads (23 bp) and try to get all hits for each read up to a specific number of mismatches (no gaps and full sensitivity). An example call would be:

```
razers3 -fl pigeonhole -ng -tc 8 -i 68 -rr 100 -m 10000000 -o output.sam hg19.fa input.fasta
```
As according to the manual RazerS3 can deal with up to 50% error rate, I wanted to detect mismatches up to 7 (percent-identity 68 for example).

My results (number of records in output SAM file) when searching for different percent identity values were the following (when I say mismatches I mean a percent identity that resolves in that specific number of mismatches):

- 5 mismatches and pigeonhole filter: 550518 records (523720 unique)
- 5 mismatches and swift filter: 26373 records (23701unique)

- 6 mismatches and pigeonhole filter: 998693 records (948911 unique)
- 6 mismatches and swift filter: 69112 records (61978 unique)

- 7 mismatches and pigeonhole filter: No records at all
- 7 mismatches and swift filter: 133854 records (118977 unique)

The main issues I see here are:

1) Both filters should report the same hits (as far as I understand the manual and publication)
2) I have a control file with experimentally determined hits (that I also rechecked with the genome) that lie within the set number of mismatches but are not detected by RazerS3 although 100% sensitivity should include all possible hits
3) There are many duplicates reported (they don't appear with the old serial version)

It would be great to have some advice on whether I missed some options or if there is any other idea why this happens.

**Comment by h-2** [comment]
@weese can you comment on this?

**Comment by rrahn** [comment]
Hi at the moment the core developer of Razers3 is no available anymore. Can you give us access to your data via an ftp share or something. We can than investigate this issue. But please bare with us, since we first have to transfer the ownership of the code to another developer.

**Comment by xp3i4** [comment]
I'm not sure the details of the tests here.
But the possible reason I think is  Razsers3 was not optimized for such short length and large error rate of reads in the tests since benchmark from its paper were almost for 100bp high quality reads. 
Besides different filters may have different results, I think.

**Comment by sarahet** [comment]
I don't think that the different filters should give different results. First of all, the read mapper should have full sensitivity according to my parameter settings and full sensitivity is just all matches - no matter what filter is used. Besides, the documentation states that the two filters are just more optimized concerning runtime for either short reads or long reads.

**Comment by xp3i4** [comment]
Hi Sara, please consider the decoys/false positive.  Full sensitivity doesn't  mean same results. 
Yes,  but what is the limitation of its length? I remember pigeonhole search qgram where q<=lengh/error+1 in your case, q<=3.  Such short length is not efficient to use pigeonhole.  So I think Razers3 is not designed for such task.
